tutorial.md

```diff
@@ -0,0 +1,212 @@
+    ).await?;
+    
+    // Initialize client with explicit credentials
+    let client = SecretManagerServiceClient::builder()
+        .with_credentials(credentials)
+        .build()
+        .await?;
+    
+    Ok(client)
+}
+```
+
+## Using the client libraries
+
+### Basic operations
+
+#### Listing secrets
+
+```rust
+use google_cloud_secretmanager_v1::client::SecretManagerServiceClient;
+
+async fn list_secrets(project_id: &str) -> Result<(), Box<dyn std::error::Error>> {
+    // Initialize the client
+    let client = SecretManagerServiceClient::new().await?;
+    
+    // Format the parent resource name
+    let parent = format!("projects/{}", project_id);
+    
+    // List all secrets in the project
+    let response = client
+        .list_secrets()
+        .set_parent(parent)
+        .send()
+        .await?;
+    
+    // Process the response
+    for secret in response.secrets {
+        println!("Secret: {}", secret.name);
+    }
+    
+    Ok(())
+}
+```
+
+### Working with long-running operations
+
+Some Google Cloud operations may take a long time to complete. Here's how to handle them:
+
+```rust
+use google_cloud_speech_v2::client::SpeechClient;
+use google_cloud_gax::retry::ExponentialBackoff;
+use std::time::Duration;
+
+async fn handle_long_running_operation(
+    project_id: &str
+) -> Result<(), Box<dyn std::error::Error>> {
+    // Initialize client with custom polling policy
+    let backoff = ExponentialBackoff::new(
+        Duration::from_secs(1),    // Initial delay
+        Duration::from_secs(60),   // Maximum delay
+        1.5,                       // Multiplier
+    );
+    
+    let client = SpeechClient::builder()
+        .with_polling_backoff_policy(backoff)
+        .build()
+        .await?;
+    
+    // Start a long-running operation
+    let operation = client
+        .create_recognizer()
+        .set_parent(format!("projects/{}/locations/global", project_id))
+        .set_recognizer_id("my-recognizer")
+        .send()
+        .await?;
+    
+    // Wait for completion
+    let result = operation.await_complete().await?;
+    
+    Ok(())
+}
+```
+
+### Error handling
+
+Handle errors appropriately to ensure your application is robust:
+
+```rust
+use google_cloud_gax::error::{Code, ServiceError};
+
+async fn robust_error_handling(
+    project_id: &str
+) -> Result<(), Box<dyn std::error::Error>> {
+    let client = match SecretManagerServiceClient::new().await {
+        Ok(client) => client,
+        Err(e) => {
+            eprintln!("Failed to initialize client: {}", e);
+            return Err(e.into());
+        }
+    };
+
+    let secret_name = format!("projects/{}/secrets/my-secret", project_id);
+    
+    match client.get_secret().set_name(secret_name).send().await {
+        Ok(secret) => {
+            println!("Retrieved secret: {}", secret.name);
+            Ok(())
+        }
+        Err(e) => {
+            if let Some(status) = e.downcast_ref::<ServiceError>() {
```

Comment:
The type of `status` would be `ServiceError`, but you are using it as if it was `Status`.... does this code compile?